File: RecommendationSystem/src/com/recommendation/utils/FileUtils.scala
package com.recommendation.utils

import java.io.BufferedReader
import java.io.FileInputStream
import java.util.zip.GZIPOutputStream
import java.io.BufferedWriter
import java.io.OutputStreamWriter
import java.io.FileOutputStream
import java.io.FileNotFoundException

object FileUtils {
  
  
  def writeToFile(filePath : String , item : String){
    val writer = new BufferedWriter(new OutputStreamWriter(new GZIPOutputStream(new FileOutputStream(filePath,true))))
    try{
      writer.write(item)
    }catch{
      case e : FileNotFoundException =>
    }finally{
      writer.close()
    } 
  }

}

File: RecommendationSystem/src/FileUtils/FileUtils.java
package FileUtils;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.util.ArrayList;
import java.util.List;

/**
 * Utility class that provides methods to read or write into files
 * 
 */


public class FileUtils {
	public void writetoFile(String filepath, String item) {
		try {

			BufferedWriter out = new BufferedWriter(new FileWriter(filepath,
			        true));
			
				out.write(item);

			out.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	public List<String> ReadFile(String filename) {
		List<String> fileContent = new ArrayList<String>();
		try {
			BufferedReader input = new BufferedReader(new FileReader(filename));
			String oneLine = null;
			while ((oneLine = input.readLine()) != null) {
				fileContent.add(oneLine);
			}
			input.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
		return fileContent;
	}
}
